services/blockchain-indexer/shared/dataService/business/token.js

```diff
@@ -56,7 +56,9 @@ const getTokenBalances = async (params) => {
 		if (response.balances) tokensInfo.push(...response.balances);
 	}
 
-	tokens.data = tokensInfo.slice(params.offset, params.offset + params.limit);
+	tokens.data = params.offset && params.limit
```

Comment:
What if we just pass either limit or offset and not both? In that case we might be returning entire tokenInfo array.

Reply:
True, wanted to keep the changes minimal, thus did not handle all the cases. When we invoke using gateway we will have the default values, modified this only to suite the case when we do not pass anything.